### PR TITLE
[ARCTIC-778][AMS]Change table format options when creating catalogs.

### LINKED ISSUE #778: [Improvement]: Change the table format option when creating catalog on AMS Dashboard
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Arctic support 3 table format now:

- Iceberg format
- Mixed Hive format
- Mixed Iceberg Format

We should change the format option on AMS Dashboard to match the definition above.

### How should we improve?

Change the table format option to :

- Iceberg
- Mixed Hive
- Mixed Iceberg

when creating a catalog.

Arctic Metastore supports Mixed Iceberg format only.
Hive Metastore supports Iceberg and Mixed Hive format only.
Hadoop and Custom support Iceberg format only.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

*Resolve #778.*

## Brief change log

  - *Change table formats to : ICEBRG、MIXED_ICEBERG、MIXED_HIVE*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Review by huiyuanz — APPROVED** [review]

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java [hunk 1]
```diff
@@ -445,7 +464,8 @@ private FilesStatistics collectOptimizeFileInfo(Collection<BaseOptimizeTask> tas
     return builder.build();
   }
 
-  private void tryUpdateOptimizeInfo(TableOptimizeInfo.OptimizeStatus optimizeStatus,
+  private void tryUpdateOptimizeInfo(
+      TableOptimizeRuntime.OptimizeStatus optimizeStatus,
                                      Collection<BaseOptimizeTask> optimizeTasks,
                                      Set<OptimizeType> types) {
```
Comment (wangtaohz) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java:470]: There is no need to use a `Set` of types

Comment (zhoujinsong) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java]: Fixed it.

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java [hunk 2]
```diff
@@ -393,49 +398,85 @@ public void updateTableOptimizeStatus() {
...
+        List<BaseOptimizeTask> fullTasks = fullPlan.plan();
+        // pending for full optimize
+        if (CollectionUtils.isNotEmpty(fullTasks)) {
+          tryUpdateOptimizeInfo(
+              TableOptimizeRuntime.OptimizeStatus.Pending, fullTasks, OptimizeType.FullMajor);
+        } else {
+          MajorOptimizePlan majorPlan = getMajorPlan(-1, System.currentTimeMillis());
+          List<BaseOptimizeTask> majorTasks = majorPlan.plan();
+          // pending for major optimize
+          if (CollectionUtils.isEmpty(majorTasks)) {
```
Comment (wangtaohz) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java:435]: `isEmpty` Should be `isNotEmpty`

Comment (zhoujinsong) [reply ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/TableOptimizeItem.java]: Fixed

**Review by wangtaohz — APPROVED** [review]
LGTM

**Pull request merged**